server/game/effects.js

```diff
@@ -1175,6 +1188,9 @@ const Effects = {
     reduceFirstPlayedCardCostEachPhase: function(amount, match) {
         return this.reduceFirstCardCostEachPhase('play', amount, match);
     },
+    reduceFirstMarshalledAmbushedOrOutOfShadowsCardCostEachPhase: function(amount, match) {
+        return this.reduceFirstCardCostEachPhase(['marshal', 'ambush', 'outOfShadows'], amount, match);
+    },
```

Comment:
I'm wondering if it's valuable to introduce these types of hyper-specific reducer effects. Most of the originals were re-used on multiple cards. Here, it feels likely that it will only be this single card.

It might be better in the long run to just improve the syntax for defining one and having it more directly in the card. 

(I'm okay if we don't do it as part of this PR)

Reply:
you are most likely right, we should remove those that are not re-used or bound to be re-used.